wagtail/contrib/search_promotions/models.py

```diff
@@ -91,26 +143,73 @@ class Meta:
 
 
 class SearchPromotion(models.Model):
+    """
+    Model for search promotion.
+
+    The promotion is for either a page or external url, not both.
+
+    Attributes
+       - query (models.ForeignKey): Matching query object.
+       - page (wagtail.models.Page): Matching page, if any.
+       - external_link_url (models.URLField): Matching external url, if any.
+       - external_link_title (models.CharField) Title text, if external url.
+       - description (models.TextField): Text description of search promotion.
+       - sort_order (int): Ordinal of search promotion.
+    """
+
     query = models.ForeignKey(
         Query, db_index=True, related_name="editors_picks", on_delete=models.CASCADE
     )
     page = models.ForeignKey(
-        "wagtailcore.Page", verbose_name=_("page"), on_delete=models.CASCADE
+        "wagtailcore.Page",
+        verbose_name=_("page"),
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
     )
-    sort_order = models.IntegerField(null=True, blank=True, editable=False)
+
+    external_link_url = models.URLField(_("URL"), blank=True)
+    external_link_title = models.CharField(
+        _("Title"), max_length=32, blank=True, null=True
+    )
```

Comment:
I’d prefer we used a more descriptive label here as well. I’d suggest word "text" rather than "title" for link text as that seems clearer to me.

```suggestion
    external_link_text = models.CharField(max_length=32, blank=True, null=True)
```

Reply:
Agree, this change still needs to be made.

Reply:
Can this please change to the suggested field name?

Reply:
"external_link_title" seems like a more appropriate name than "external_link_text" for 2 reasons: 1) the field is used in the "title" property when the search promotion is an external link instead of a page; 2) it is a char field, not a text field. An enhancement someone might like to make is adding a button the lets wagtail get the title for the URL the user entered and automatically fill in the field.

Reply:
Personally I agree with Thibaud's original suggestion, here's my reasoning.

1. We already call this similar think 'text' when shown an external link in the link chooser modal. 
2. Using the term title could confuse the user into thinking this is a link title (as in the html attribute).
3. If we ever want to add support for link title (unlikely but possible), we now need a migration.
4. It's possible we would add support for link target though, which IS an attribute, hence we would have title (not the attribute title but the link text/content) and target (which would be the attribute).
5. Having this a bit more generic leaves it up to implementors how they want to present this in their own search results list.

```html
<a href="https://www.example.com/contact-us" title="click to load the contact us page" target="_blank">Contact</a>
```

Reply:
Assuming 2 people's opinions make a consensus, we've made the change from "external_link_title" to "external_link_text". It might be a good idea for the wagtail team to clarify what they consider a consensus.